File: frontend/src/services/implementations/weather_service.ts
import { IWeatherService } from '../interfaces/Iweather_service';
import { WeatherForecast, WeatherInfo } from '../schemas/weather_resp_schema';

export class WeatherService implements IWeatherService {
    private baseUrl: string;

    constructor(baseUrl: string = 'http://localhost:8000/api/v1') {
        this.baseUrl = baseUrl;
    }

    async get_current_weather(region: string): Promise<WeatherInfo> {
        const response = await fetch(`${this.baseUrl}/weather/${region}`);
        if (!response.ok) {
            throw new Error('Failed to call backend');
        }
        return response.json();
    }

    async get_weather_forecast(region: string, days: number): Promise<WeatherForecast> {
        const response = await fetch(`${this.baseUrl}/weather/forecast/${region}?day=${days}`);
        console.log(response);
        if (!response.ok) {
            throw new Error('Failed to call backend');
        }
        return response.json();
    }
}

export class WeatherService1 implements IWeatherService {
    private baseUrl: string;

    constructor(baseUrl: string = 'http://localhost:8000/api/v1') {
        this.baseUrl = baseUrl;
    }

    async get_current_weather(region: string): Promise<WeatherInfo> {
        const response = {
            region: region,
            temperature: 25,
            condition: 'Sunny',
            humidity: 60
        };
        return response as WeatherInfo;
    }

    async get_weather_forecast(region: string, days: number): Promise<WeatherForecast> {
        const response = await fetch(`${this.baseUrl}/weather/forecast/${region}?day=${days}`);
        if (!response.ok) {
            throw new Error('Failed to call backend');
        }
        return response.json();
    }
}

File: backend/src/backend/services/implementation/weather_service.py
from typing import Dict, Any
from backend.services.interfaces.Iweather_service import IWeatherService

from backend.services.schemas.weather_resp_schema import WeatherForecastResponse, WeatherResponse


class WeatherService(IWeatherService):
    async def get_current_weather(self, region_name: str) -> WeatherResponse:
        weather_data = {
            "region": region_name,
            "temperature": 22,
            "condition": "Ensoleillé",
            "humidity": 65
        }
        return WeatherResponse(**weather_data)


class WeatherService1(IWeatherService):
    async def get_current_weather(self, region_name: str) -> WeatherResponse:
        response = {
        "region": "le nom que j'ai choisit",
        "temperature": 1000,
        "condition": "ça va",
        "humidity": 20
        }
        return WeatherResponse(**response)
    
    async def get_weather_forecast(self, region_name, days) -> WeatherForecastResponse:
        response = {
            "region": region_name,
            "temperature": 0,
            "condition": "ça va dans les prochains jours",
            "humidity": 20,
            "day": days
        }
        return WeatherForecastResponse(**response)